python/src/py_stream/py_stream.rs

```diff
@@ -22,6 +26,86 @@ impl PyStream {
     fn id(&self) -> String {
         format!("{}", self.stream.id())
     }
+
+    fn map(&self, map_fn: PyObject) -> PyStream {
+        let new_fn = Arc::new(map_fn);
+        let f = move |data: &Vec<u8>| -> Vec<u8> {
+            Python::with_gil(|py| {
+                let serialized_data = PyBytes::new(py, &data[..]);
+                let pickle = PyModule::import(py, "pickle").unwrap();
+                let des_data: PyObject = pickle
+                    .getattr("loads")
+                    .unwrap()
+                    .call1((serialized_data,))
+                    .unwrap()
+                    .extract()
+                    .unwrap();
+                let result: PyObject = new_fn.call1(py, (des_data,)).unwrap().extract(py).unwrap();
+                pickle
+                    .getattr("dumps")
+                    .unwrap()
+                    .call1((result,))
+                    .unwrap()
+                    .extract()
+                    .unwrap()
```

Comment:
It might be cleaner to do serialization/deserialization in a Python wrapper function instead of in Rust. I realize we're doing the same thing in the Python operators, so fine with keeping this as is for this PR, but we may want to make a follow-up PR to clean up serialization/deserialization.
@sukritkalra what do you think?